Sources/SmokeHTTP1/SmokeInwardsRequestContext.swift

```diff
@@ -31,22 +31,55 @@ internal class StandardSmokeInwardsRequestContext: SmokeInwardsRequestContext, O
     private(set) var retriableOutputRequestRecords: [RetriableOutputRequestRecord]
     private(set) var retryAttemptRecords: [RetryAttemptRecord]
     
+    internal let accessQueue = DispatchQueue(
```

Comment:
Is it better to use `Actor`s instead of managing the queue ourselves?

Reply:
Yes ideally but the internals of smoke-framework and smoke-http are still either completion or EventLoopFuture based so it would be difficult to integrate an actor.

For the next major version or if we create a new client based on middleware, we should definitely make this an actor.
https://github.com/amzn/smoke-http/blob/middleware_poc/Sources/SmokeHTTPClientMiddleware/RetriableOutwardsRequestAggregator.swift